R/utils-validation.R

```diff
@@ -0,0 +1,75 @@
+#' Accumulate Check Results
+#'
+#' @param severity (`string`)\cr
+#'   A severity name to provide, expecting `"warn"` or `"stop"`
+#' @param cond (`bool`)\cr
+#'   A condition upon which the message should be emitted
+#' @param ...
+#'   A message to emit
+#' @param envir
+#'   The environment in which a hidden `..checks..` object should be
+#'   accumulated. Defaults to the parent frame.
+#'
+#' @return The condition used to decide whether to emit a message
+#'
+#' @keywords internal
+check <- function(severity, cond, ..., envir = parent.frame()) {
```

Comment:
let's not go into this here pls. We can first have a discussion issue on this kind of stuff.

Reply:
Yep - see my TODO. I've since replaced it with `checkmate::makeAssertCollection`. I'll circle back to the rest of your comments after I wrap the the TODOs, but I think I've already addressed the majority.